### PR TITLE
Initial instrumentation

### PR DESCRIPTION
**Pull request by charlescolbourn**
### What

Included instrumentation for OT trace reporting

### How to review

Changes should have no impact without environment variables set and a collector present. With required infrastructure traces should appear in UI.

### Who can review

Fran Moore

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -123,7 +140,7 @@ func main() {
 		NewDatasetRoutingEnabled:     cfg.NewDatasetRoutingEnabled,
 		PrefixDatasetHandler:         prefixDatasetHandler,
 		DatasetClient:                datasetClient,
-		HealthCheckHandler:           hc.Handler,
+		// HealthCheckHandler:           hc.Handler,
```
Comment (franmoore05) [inline main.go:143]: Is there any reason the health check handler has been removed or will this be uncommented when ready?

File: main.go [hunk 2]
```diff
@@ -80,11 +97,11 @@ func main() {
 	// Healthcheck API
 	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
 	if err != nil {
-		log.Fatal(ctx, "Failed to obtain VersionInfo for healthcheck", err)
+	log.Fatal(ctx, "Failed to obtain VersionInfo for healthcheck", err)
```
Comment (franmoore05) [inline main.go:100]: Could the indentation be replaced please as this will likely fail linting checks otherwise.

File: main.go [hunk 3]
```diff
@@ -80,11 +97,11 @@ func main() {
...
 	}
 	hc := healthcheck.New(versionInfo, cfg.HealthcheckCriticalTimeout, cfg.HealthcheckInterval)
 	if err = hc.AddCheck("API router", zebedeeClient.Checker); err != nil {
-		log.Fatal(ctx, "Failed to add api router checker to healthcheck", err)
+	log.Fatal(ctx, "Failed to add api router checker to healthcheck", err)
```
Comment (franmoore05) [inline main.go:104]: Could the indentation be replaced please as this will likely fail linting checks otherwise.

File: main.go (hunk 1 above)
Comment (charlescolbourn) [reply main.go:143]: my fault. I commented it out to reduce log noise. It's back in locally but I haven't pushed the change yet.

**Review by franmoore05 — APPROVED** [review]

**Review by franmoore05 — APPROVED** [review]

**Pull request merged**